connectors/sources/confluence.py

```diff
@@ -152,6 +171,40 @@ async def close_session(self):
         await self.session.close()
         self.session = None
 
+    async def _handle_client_errors(self, url, exception):
+        if exception.status == 429:
+            response_headers = exception.headers or {}
+            retry_seconds = DEFAULT_RETRY_SECONDS
+            if "Retry-After" in response_headers:
+                try:
+                    retry_seconds = int(response_headers["Retry-After"])
+                except (TypeError, ValueError) as exception:
+                    self._logger.error(
+                        f"Error while reading value of retry-after header {exception}. Using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
+                    )
+            else:
+                self._logger.warning(
+                    f"Rate Limited but Retry-After header is not found, using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
+                )
+            self._logger.debug(f"Rate Limit reached: retry in {retry_seconds} seconds")
+
+            await self._sleeps.sleep(retry_seconds)
+            raise ThrottledError
+        elif exception.status == 404:
+            self._logger.error(f"Getting Not Found Error for url: {url}")
+            raise NotFound
+        elif exception.status == 500:
+            self._logger.error("Internal Server Error occurred")
+            raise InternalServerError
+        else:
+            raise
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
```

Comment:
As you move the retry logic to `@retryable`, you should remove `while True` as well.